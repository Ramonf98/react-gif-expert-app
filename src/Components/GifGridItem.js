import React from 'react'
// import PropTypes from 'prop-types'

const GifGridItem = ( { title, url} ) => {
    return (
        <div className="gifCard animate__animated animate__fadeIn ">
            <img 
            className="gif" 
            src={url} 
            alt="Gif No cargado">
            </img>
            
            <p className="gifTitle">
                {title}
            </p>
        </div>
    )
}

export default GifGridItem

