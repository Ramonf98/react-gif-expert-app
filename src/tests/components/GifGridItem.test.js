import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../Components/GifGridItem'; //Componente a probar
import '@testing-library/jest-dom'; //Para ayuda con autocompletado


describe('Pruebas en el archivo GifGridItem.js',()=>{
    const title = 'Batman';
    const url = 'https://www.google.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Deben mostrar <GifGridItem /> correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    })


    test('El title debe estar en un p, parrafo ', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('El src de la imagen debe ser igual a la url, y el alt debe ser igual "Gif No cargado" ', () => {
        
        const img = wrapper.find('img');
        const {src, alt} = img.props();
        expect(src).toBe( url );
        expect(alt).toBe('Gif No cargado');
        
    })

    test('Debe tener la clase animate_fadeIn ', () => {
        const div = wrapper.find('div');
        
        expect( div.hasClass('animate__fadeIn') ).toBeTruthy();
    })
    

})