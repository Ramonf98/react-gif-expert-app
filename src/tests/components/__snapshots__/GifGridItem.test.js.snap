// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GifGridItem.js Deben mostrar <GifGridItem /> correctamente  1`] = `
<div
  className="gifCard animate__animated animate__fadeIn "
>
  <img
    alt="Gif No cargado"
    className="gif"
    src="https://www.google.com"
  />
  <p
    className="gifTitle"
  >
    Batman
  </p>
</div>
`;
